src/transports/http.ts

```diff
@@ -467,6 +467,35 @@ class TransportHTTP implements TransportService {
     });
   }
 
+  /**
+   * Register a new paymail
+   * @param {string} key - The rawXPubKey
+   * @param {string} address - The full paymail address
+   * @param {string} [publicName] - The public name (optional)
+   * @param {string} [avatar] - The avatar (optional)
+   * @param {Metadata} [metadata] - The metadata to record on the destination (optional)
+   * @returns {null}
+   */
+  async NewPaymail(
+    key: string,
+    address: string,
+    publicName?: string,
+    avatar?: string,
+    metadata?: Metadata
+  ): Promise<null> {
+    await this.doHTTPRequest(`${this.serverUrl}/paymail`, {
+      method: "POST",
+      body: JSON.stringify({
+        key,
+        address,
+        public_name: publicName,
+        avatar,
+        metadata,
+      }),
+    });
+    return null;
```

Comment:
This method on a server returns `201 Created` success response or `422 Unprocessable Entity` error response. Maybe we could return that response from that function as well instead of returning `null`?

Reply:
Actually doHTTPRequest will throw an Error when there is a status code >=400 so the Promise will fail, so there is no need to return anything.

But maybe it's better to set a return type as a `Promise<void>`?

Reply:
@kuba-4chain @dorzepowski 

In my implementation, this function returns `null` on success, as this best reflects the behaviour of the corresponding function in `go-buxclient`, which effectively returns `nil` or an error.

But I agree, here in `js-buxclient` it's a good idea to change it so that `Promise<void>` is returned.